test/e2e/upgrade_test.go

```diff
@@ -73,27 +52,20 @@ func upgradeTestSuite(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
-	// generate node affinities
-	affinities, err := getNodeTypeAffinities()
-	require.NoError(t, err, "error getting node affinity")
-	// loop affinities to ensure deployed workloads will exclusively run based
-	// on the given affinity, e.g.  BYOH and/or Machine nodes
-	for _, affinity := range affinities {
-		// test if Windows workloads are running by creating a Job that curls the workloads continuously.
-		deployment, err := testCtx.deployWindowsWebServer("win-webserver", affinity)
-		require.NoErrorf(t, err, "error creating Windows Webserver deployment for upgrade test")
-		defer testCtx.deleteDeployment(deployment.Name)
-
-		// create a clusterIP service which can be used to reach the Windows webserver
-		intermediarySVC, err := testCtx.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
-		require.NoErrorf(t, err, "error creating service for deployment %s", deployment.Name)
-		defer testCtx.deleteService(intermediarySVC.Name)
-
-		// create a Job object that continuously curls the webserver every 5 seconds.
-		testerJob, err := testCtx.createLinuxJob(windowsWorkloadTesterJob, generateLinuxWorkloadTesterCommand(intermediarySVC.Spec.ClusterIP))
-		require.NoErrorf(t, err, "error creating linux job %s", windowsWorkloadTesterJob)
-		defer testCtx.deleteJob(testerJob.Name)
-	}
+	// test if Windows workloads are running by creating a Job that curls the workloads continuously.
+	deployment, err := testCtx.deployWindowsWebServer("win-webserver", nil)
+	require.NoErrorf(t, err, "error creating Windows Webserver deployment for upgrade test")
+	defer testCtx.deleteDeployment(deployment.Name)
+
+	// create a clusterIP service which can be used to reach the Windows webserver
+	intermediarySVC, err := testCtx.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+	require.NoErrorf(t, err, "error creating service for deployment %s", deployment.Name)
+	defer testCtx.deleteService(intermediarySVC.Name)
+
+	// create a Job object that continuously curls the webserver every 5 seconds.
+	testerJob, err := testCtx.createLinuxJob(windowsWorkloadTesterJob, generateLinuxWorkloadTesterCommand(intermediarySVC.Spec.ClusterIP))
+	require.NoErrorf(t, err, "error creating linux job %s", windowsWorkloadTesterJob)
+	defer testCtx.deleteJob(testerJob.Name)
```

Comment:
Please revert back to how the code was before i.e. reintroduce `deployWindowsWorkloadAndTester()`. Sorry this is not concise code but belongs to a method that by looking at the signature one can understand what it does. I had called this out [earlier](https://github.com/openshift/windows-machine-config-operator/pull/645#discussion_r710261878).

Reply:
Reverted in https://github.com/openshift/windows-machine-config-operator/pull/674/commits/d3b3c0c2bdc42999910ff263994ae8516d85044e